src/Promitor.Core/Metrics/MeasuredMetric.cs

```diff
@@ -63,9 +64,14 @@ public static MeasuredMetric CreateForDimension(double? value, string dimensionN
         {
             Guard.NotNullOrWhitespace(dimensionName, nameof(dimensionName));
             Guard.NotNull(timeseries, nameof(timeseries));
-            Guard.For<ArgumentException>(() => timeseries.Metadatavalues.Any() == false);
+            
+            var dimensionMetadataValue = timeseries.Metadatavalues.Where(metadataValue => metadataValue.Name?.Value.Equals(dimensionName, StringComparison.InvariantCultureIgnoreCase) == true);            
+            if(!dimensionMetadataValue.Any())
+            {
+                throw new MissingDimensionException(dimensionName);
```

Comment:
Any reason why we did not add the timeseries to the exception?

Reply:
I wasn't sure how I'd want to string-fy the timeseries in the exception message, besides just converting it to JSON; we already do that in a debug log though. 

I think it does make sense though to add the time series to the exception, just as a field one can access. Updating

Reply:
Thanks!